### PR TITLE
Fix global style for Product Summary block, Product Stock Indicator block, and Product Title block

### PR DESCRIPTION
**Pull request by gigitux**
This PR fixes the global style for `Product Summary block`, `Product Stock Indicator block`, and `Product Title block` when these blocks are used in the `All Products` block.

This code register the `Product Summary block`, `Product Stock Indicator block`, and `Product Title` block on the PHP side. This is necessary for using the `__experimentalSelector` from global style API.

### Screenshots

image
*with global style*

image
*without global style*

### Manual Testing

How to test the changes in this Pull Request:

1.  Upgrade to `WordPress 5.9`.
2.  Install and enable the `Twenty Twenty-Two` theme.
3.  Add the `All Product Block` (this block contains `Stock Indicator block`) to a post.
4.  Click on the pencil to edit the block, add the `Stock Indicator Block` and get the focus on the `Stock Indicator block`.
5.  On the right sidebar, personalize the styles of the block.
6.  Go on the page and check if there are changes.
7.  Reset to default using the `Reset` button from the different sections.
8.  Go to Dashboard and select Appearance > Editor (beta). On top of the screen, select Home > Browser all templates > Single Post. When the page is loaded, add the block to the page. Click on the pencil to edit the block, add the `Stock Indicator Block`.
9.  On the Editor page click on the `Styles` icon on the right-top corner.
10. Verify that the `Stock Indicator block` is shown under the `Blocks` section. Personalize the block.
11. Save your changes.
12. Go on the page created earlier and check if all styles are applied correctly.
13. Edit your previous post/page again.
14. Change again the styles.
15. Save your changes.
16. Check if these styles have priority over the styles from the Site Editor.

Repeat from 4th point for `Product Summary Block` and `Product Title Block`.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
* [x] Same as above

### CODE REVIEW

**Review by Aljullu — APPROVED** [review]
Looks good, thanks for fixing this @gigitux!

I noticed there are some global styles that are not applied in the All Products frontend. Ie, testing the Product Title block:

| Style editor | Frontend |
| --- | --- |
| imatge | imatge |

Those seem to be:
* Text color
* Font family
* Font size
* Line height
* Font weight
* Margin

Can you reproduce it too? I don't think it's blocking release, but we should create an issue and fix it in a future release.

I also found another two issues (#5596 and #5597), but they only affect dev builds so not blocking the release at all. However, if you think they might be regressions after the work done in the global styles project, it might make sense adding them to the epic.

File: src/BlockTypes/ProductStockIndicator.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	 *
+	 * @return array
+	 */
+	protected function get_block_type_supports() {
+		return array(
+			'color'                  =>
+			array(
+				'link'  => false,
+				'background' => false,
+				'text' => true,
```
Comment (nielslange) [inline src/BlockTypes/ProductStockIndicator.php]: Based on the PHP Coding Standard, we align `=` and `=>` under each other to increase the readability. You can see an example of that on https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#indentation. 

Don't worry, there's no need to manually do that. `npm run lint:php-fix` will do that job for you. 😉

File: src/BlockTypes/ProductSummary.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	 *
+	 * @return array
+	 */
+	protected function get_block_type_supports() {
+		return array(
+			'color'                  =>
+			array(
+				'link'  => true,
+				'background' => false,
+				'text' => true,
```
Comment (nielslange) [inline src/BlockTypes/ProductSummary.php]: Same as the comment before.

**Review by nielslange — COMMENTED** [review]
Functionality-wise, the PR works as expected. 👏 I only noticed two minor points:

1. The `Product Title` cannot be aligned due to https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/345bfcd0b83c6cb5c71a66fc3e402352a23a8e1e/assets/js/atomic/blocks/product-elements/title/style.scss#L1-L3
2. In PHP, `=` and `=>` of a code block should be aligned according to https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#indentation. I left a note that explains how to fix that.

**Review by nielslange — APPROVED** [review]
LGTM! 🚢

**Comment by manospsyx** [comment]
@gigitux I'm seeing some 404s related with these blocks when editing widget areas -- is it likely that they are [registered there](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/557cbbc18a1b0db1ad6482abb695996a642bf3af/src/BlockTypesController.php#L200-L208)?

Screen Shot 2022-01-23 at 12 29 06 AM

**Comment by gigitux** [comment]
@manospsyx do you still be able to reproduce this issue? I'm not able to see these errors. I'm on WordPress 5.9.
I installed [Classic Widgets plugin](https://wordpress.org/plugins/classic-widgets/) and navigated to `/wp-admin/widgets.php`, but I didn't see any HTTP error

**Comment by manospsyx** [comment]
Hey @gigitux -- I still see those when navigating to `wp-admin/widgets.php` (WP 5.9, Blocks built from `trunk`). Not sure if it helps but I'm not running the Classic Widgets Plugin. @jimjasson were you also seeing those?

**Comment by gigitux** [comment]
@manospsyx Ok, without `Classic Widgets` I'm able to see the errors. I'm already working on the fix, thanks!

**Comment by gigitux** [comment]
The fix is ready! #5649 

@manospsyx, thanks for reporting it!

**Pull request merged**